### PR TITLE
RRule: Register end time using `Rule::setUntil()` instead

### LINKED ISSUE #35: Generated RRULE is invalid once an end date is set
`DTEND` is not valid, `UNTIL` must be used instead. Other rrule libraries like [rrule.js](https://jakubroztocil.github.io/rrule/) cannot parse the RRULE otherwise.

### PR DESCRIPTION
**Pull request by nilmerg**
fixes #35

### CODE REVIEW

File: tests/RRuleTest.php [hunk 1]
```diff
@@ -156,4 +156,20 @@ public function testJsonSerializeAndDeserialize()
 
         $this->assertEquals($rrule, RRule::fromJson(json_encode($rrule)));
     }
+
+    public function testRecurrenceEndIsProperlySet()
+    {
+        $rrule = RRule::fromFrequency(RRule::DAILY)
+            ->startAt(new DateTime('2023-01-01T12:00:00', new \DateTimeZone('UTC')))
+            ->endAt(new DateTime('2024-01-01T12:00:00', new \DateTimeZone('UTC')));
```
Comment (yhabteab) [inline tests/RRuleTest.php:164]: Can you please import this.

**Review by yhabteab — CHANGES_REQUESTED** [review]

File: tests/RRuleTest.php [hunk 2]
```diff
@@ -156,4 +156,20 @@ public function testJsonSerializeAndDeserialize()
...
+
+        $this->assertSame(
+            [
+                'start'     => '2023-01-01T12:00:00+00:00',
+                'rrule'     => 'FREQ=DAILY;UNTIL=20240101T120000Z',
+                'frequency' => RRule::DAILY
+            ],
+            $rrule->jsonSerialize()
```
[the remark this answers is not in the document]
Comment (nilmerg) [reply tests/RRuleTest.php:172]: The intention for this test, is to verify that `UNTIL` is used, not `DTEND`. Hence this must be done this way.

[the remark this answers is not in the document]
Comment (yhabteab) [reply tests/RRuleTest.php:172]: But if you don't set the `DTEND` with `setEndDate()`, which is not the case here, it won't be used in any way.

[the remark this answers is not in the document]
Comment (nilmerg) [reply tests/RRuleTest.php:173]: Please lookup what a regression test is. Then you probably understand what I want to achieve.

**Review by yhabteab — APPROVED** [review]

**Pull request merged**